File: superawesome-base/src/test/java/tv/superawesome/lib/ResourceReader.java
package tv.superawesome.lib;

import java.io.InputStream;
import java.util.Scanner;

/**
 * Created by gabriel.coman on 30/04/2018.
 */
public class ResourceReader {1

    public static String readResource(String name) {
        InputStream in = ClassLoader.getSystemClassLoader().getResourceAsStream(name);
        return convertStreamToString(in);
    }

    private static String convertStreamToString(InputStream is) {
        String kCHARSET = "UTF-8";
        String kDELIMITER = "\\A";
        Scanner s = new Scanner(is, kCHARSET).useDelimiter(kDELIMITER);
        return s.hasNext() ? s.next() : "";
    }
}

File: superawesome-common/src/test/java/tv/superawesome/sdk/publisher/common/testutil/ResourceReader.kt
package tv.superawesome.sdk.publisher.common.testutil

import java.io.InputStream
import java.util.Scanner

object ResourceReader {
    fun readResource(name: String?): String {
        val inputStream = ClassLoader.getSystemClassLoader().getResourceAsStream(name)
        return convertStreamToString(inputStream)
    }

    private fun convertStreamToString(inputStream: InputStream): String {
        val kCHARSET = "UTF-8"
        val kDELIMITER = "\\A"
        val s = Scanner(inputStream, kCHARSET).useDelimiter(kDELIMITER)
        return if (s.hasNext()) s.next() else ""
    }
}
